### PR TITLE
YLP-1868 Rewrote teams route and invite members route to use new ones from bff

### PR DESCRIPTION
**Pull request by TimCanu**
- Inviting a member also returns the team list
- The teams returned are ready to be used and no longer need to be mapped

### CODE REVIEW

File: packages/yourloops/lib/team/team-api.ts [hunk 1]
```diff
@@ -62,9 +66,20 @@ interface RemoveMemberArgs {
...
 export default class TeamApi {
-  static async getTeams(): Promise<ITeam[]> {
+  static getTeamsApiUrl(user: User): string {
+    if (user.isUserHcp()) {
+      return `/bff/v1/hcps/${user.id}/teams`
+    } else if (user.isUserPatient()) {
+      return `/bff/v1/patients/${user.id}/teams`
+    } else {
+      throw Error(`User with role ${user.role} cannot retrieve teams`)
+    }
```
Comment (celia-rouis) [inline packages/yourloops/lib/team/team-api.ts:76]: Some improvements could be done to this method:
- The error case could be handled first
- The HCP and patient case could be factorized

```ts
// To put at the top of the file to make sure these `const` never change
const HCP_ROUTE = 'hcps'
const PATIENTS_ROUTE = 'patients'

// ...
static getTeamsApiUrl(user: User): string {
  if (!user.isUserHcp() && !user.isUserPatient() {
    throw Error('...')
  }
  const userRoute = user.isUserHcp() ? HCP_ROUTE : PATIENT_ROUTE
  return `/bff/v1/${userRoute}/${user.id}/teams`
}

```

**Review by fatb38 — CHANGES_REQUESTED** [review]

File: packages/yourloops/components/team/member-row.tsx [hunk 2]
```diff
@@ -97,7 +97,7 @@ function MemberRow(props: TeamMembersProps): JSX.Element {
     (loggedInUserId === currentUserId && TeamUtils.isUserTheOnlyAdministrator(team, loggedInUserId)) ||
     userUpdateInProgress
   const removeMemberDisabled = !loggedInUserIsAdmin || userUpdateInProgress || loggedInUserId === currentUserId ||
-    (teamMember.status === UserInvitationStatus.pending && (!teamMember.invitation || team.id !== teamMember.invitation.target?.id)) // This condition basically means that the logged in user did not invite the pending user
+    (teamMember.status === UserInvitationStatus.pending && !teamMember.invitationId) // This condition basically means that the logged in user did not invite the pending user
```
[the remark this answers is not in the document]
Comment (TimCanu) [reply packages/yourloops/components/team/member-row.tsx:100]: Yes if you did not invite him yourself

File: packages/yourloops/lib/team/team-api.ts [hunk 3]
```diff
@@ -61,10 +65,23 @@ interface RemoveMemberArgs {
...
 
+const HCP_ROUTE = 'hcps'
+const PATIENTS_ROUTE = 'patients'
+
 export default class TeamApi {
-  static async getTeams(): Promise<ITeam[]> {
+  static getTeamsApiUrl(user: User): string {
+    const isUserHcp = user.isUserHcp()
+    if (!isUserHcp && !user.isUserPatient()) {
+      throw Error(`User with role ${user.role} cannot retrieve teams`)
```
[the remark this answers is not in the document]
Comment (TimCanu) [reply packages/yourloops/lib/team/team-api.ts:75]: It the same for all the Error('...') that we throw in the app (we have 35 of them). Don't you think that this is out of scope?

File: packages/yourloops/test/unit/lib/notifications/api.test.tsx [hunk 4]
```diff
@@ -302,24 +302,24 @@ describe('Notification API', () => {
...
-        try {
-          await NotificationApi.acceptInvitation(userId, { ...notification, type: notificationType })
-        } catch (reason) {
-          err = reason as Error
-        }
-        expect(err).not.toBeNull()
+      try {
+        await NotificationApi.acceptInvitation(userId, {
+          ...notification,
+          type: 'unknownType' as unknown as NotificationType
```
[the remark this answers is not in the document]
Comment (TimCanu) [reply packages/yourloops/test/unit/lib/notifications/api.test.tsx:317]: I don't know how

File: packages/yourloops/lib/notifications/notification-api.ts [hunk 5]
```diff
@@ -49,31 +49,20 @@ export default class NotificationApi {
...
-          throw Error('Missing or invalid team ID in notification')
-        }
-        payload.target = notification.target
-        break
-      case NotificationType.directInvitation:
-        payload.email = notification.email
-        break
-      default:
-        throw new Error('Invalid notification type')
+  static async cancelInvitation(notificationId: string, teamId?: string, inviteeEmail?: string): Promise<void> {
```
[the remark this answers is not in the document]
Comment (TimCanu) [reply packages/yourloops/lib/notifications/notification-api.ts:58]: I checked the back. The route call can cancel any invite. It searches an invite in the db (based on the id given) but you must have to give either the teamId or the invitee email.
That is why, in the cases in which I have both, I prefer to give both :slightly_smiling_face:

File: packages/yourloops/test/unit/common/utils.ts [hunk 6]
```diff
@@ -182,7 +178,7 @@ export function buildTeamMember(
     profile: { fullName },
     role,
     status,
-    invitation
+    invitationId
   }
 }
```
[the remark this answers is not in the document]
Comment (TimCanu) [reply packages/yourloops/test/unit/common/utils.ts]: Good catch :slightly_smiling_face:

File: packages/yourloops/lib/team/team-api.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (fatb38) [reply packages/yourloops/lib/team/team-api.ts:77]: https://www.linkedin.com/posts/javascript-developer_activity-6995787749240811520-1jkq?utm_source=share&utm_medium=member_desktop

File: packages/yourloops/test/unit/lib/notifications/api.test.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (fatb38) [reply packages/yourloops/test/unit/lib/notifications/api.test.tsx:316]: Ok, hard work I guess, maybe later in when refactoring Notifications.

File: packages/yourloops/lib/team/team-api.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (TimCanu) [reply packages/yourloops/lib/team/team-api.ts:77]: ahaha but there is no rock in my way in my case

File: packages/yourloops/lib/notifications/notification-api.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (fatb38) [reply packages/yourloops/lib/notifications/notification-api.ts:56]: Ok, it's clear now 😉

**Comment by TimCanu** [comment]
Do not merge. I think that there is an issue with an E2E test

**Pull request merged**